docs/spec.rst

```diff
@@ -0,0 +1,498 @@
+   in `implementation
+   code <https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit#bookmark=id.cjuxrgo7keqs>`__
+   (and the semantics of doing so) is currently unspecified:
+
+   -  A local variable type.
+   -  The type in a cast expression.
+   -  An array or object creation expression.
+   -  An explicit type argument supplied to a generic method or
+      constructor (including via a member reference), or to an instance
+      creation expression for a generic class.
+
+-  Applying more than one of these annotations to the same `type
+   component <https://docs.google.com/document/d/1KQrBxwaVIPIac_6SCf--w-vZBeHkTvtaqPSU_icIccc/edit#bookmark=kix.g7gl9fwq1tt5>`__
+   within the same type usage (since we currently define only
+   ``@Nullable``, this means applying ``@Nullable`` twice) is illegal.
+   [*debated*
+   `#100 <https://github.com/jspecify/jspecify/issues/100>`__]
+
+Tools are encouraged to treat an illegally placed annotation in Java
+source code as an error. In bytecode, illegally placed annotations may
+be best ignored.
+
+Type usages defined here as *legal* or *illegal* (as opposed to
+*unspecified*) to receive type use annotations are considered **in
+scope** of this specification. Tools may extend this specification to
+*unspecified* type usages at their own choosing.
+
+The defaulting annotations
+--------------------------
+
+In this release, we will provide a single parameterless declaration
+annotation called ``@DefaultNonNull``.
+[`#5 <https://github.com/jspecify/jspecify/issues/5>`__,
+`#87 <https://github.com/jspecify/jspecify/issues/87>`__]
+
+.. _structural-legality-defaulting:
+
+Structural legality
+~~~~~~~~~~~~~~~~~~~
+
+-  The set of defaulting annotations can be applied to:
+
+   -  A *named* class.
+   -  A package. [*debated*
+      `#34 <https://github.com/jspecify/jspecify/issues/34>`__]
+   -  A module. [*debated*
+      `#34 <https://github.com/jspecify/jspecify/issues/34>`__]
+   -  *Not* a method
+      [`#43 <https://github.com/jspecify/jspecify/issues/43>`__],
+      constructor
+      [`#43 <https://github.com/jspecify/jspecify/issues/43>`__], or
+      field [`#50 <https://github.com/jspecify/jspecify/issues/50>`__].
+      *debated*
+
+-  Applying more than one of these annotations to the same scope (since
+   we currently define only ``@DefaultNonNull``, this means applying
+   ``@DefaultNonNull`` twice) is illegal. [*debated*
+   `#100 <https://github.com/jspecify/jspecify/issues/100>`__,
+   `#87 <https://github.com/jspecify/jspecify/issues/87>`__]
+
+Semantics
+---------
+
+In this section, we discuss how type checking should play out.
+
+Type hierarchy
+~~~~~~~~~~~~~~
+
+Our nullability annotations produce the following apparent type
+hierarchy [`#80 <https://github.com/jspecify/jspecify/issues/80>`__]:
+
+| ``Foo`` (written in the context of ``@DefaultNonNull``)
+| ``<: Foo`` (written outside the context of ``@DefaultNonNull``)
```

Comment:
nit: you could use "<=" (or some other symbol) to avoid overloading Java's base subtyping

Reply:
Done.